packages/@aws-cdk/aws-codebuild/lib/source.ts

```diff
@@ -482,6 +482,13 @@ interface ThirdPartyGitSourceProps extends GitSourceProps {
    */
   readonly webhook?: boolean;
 
+  /**
+   * Start a batch build when the webhook is triggered instead of a standard one.
```

Comment:
The "when the webhook is triggered" is an interesting statement. So this setting works _only_ for builds started through the webhook? Not through, for instance, CloudWatch Events?

If that is indeed the case, I'd like to change this a little bit:

1. Let's call this property `webhookTriggersBatchBuild?: boolean` instead, to make it absolutely clear what it does.
2. If `webhookTriggersBatchBuild` is passed as `true`, but `webhook` is `false` (which gets a little complicated because `webhook` is considered `true` if it's `undefined`, but `webhookFilters` have been provided), that sounds like an error to me, and we should validate that scenario (and check that logic with a separate unit test, of course 🙂).

Reply:
Hmm good question. I assume it applies for anything that triggers. Asked here: https://github.com/aws-cloudformation/aws-cloudformation-coverage-roadmap/issues/621#issuecomment-736036058

Reply:
@tjenkinson can you perhaps do an experiment with that setting and CloudWatch Events, and report back what you find? I don't want to hold your PR hostage to this question 🙂.

Reply:
Looking at the console
![image](https://user-images.githubusercontent.com/3259993/101931905-f3d5a280-3bd1-11eb-8c5d-57619710a2dd.png)


It really implies it's just webhooks?

Reply:
Looking at the docs [here](https://docs.aws.amazon.com/AmazonCloudWatch/latest/events/CloudWatch-Events-tutorial-codebuild.html) I don't think it's possible to trigger a batch build from cloudwatch, because it mentions that any options you provide are for the `StartBuild` call, not `StartBuildBatch`

>For more information about the parameters you can pass, see StartBuild. You cannot pass the projectName parameter in this field. Instead, you specify the project using the ARN in Project ARN.

Reply:
@tjenkinson OK. Let's assume that it is the case then, and let's implement the 2 points I made here https://github.com/aws/aws-cdk/pull/11743#discussion_r532747769 .

Reply:
@skinny85 currently if you pass `webhooks: false` and also provide filter groups, then the end result is no error, and the `SourceConfig` contains `webhook` `false` and the filter groups.

So I followed the same logic and decided to default `webhooks` to `true` if it was omitted and there were filter groups or `webhookTriggersBatchBuild` was `true`.

Also if you explicitly set `webhooks` to `false` we will leave it as `false` but still set the `buildType` in the `SourceConfig` just like we do with the filters.

WDYT?

Reply:
I feel like passing `webhooks: false`, but `webhookFilters` is probably a mistake...

In fact, I just tried it, and I got the following error from CodeBuild's API:

```
Failed to call CreateWebhook, reason: Cannot specify filter groups when webhook is disabled.
(Service: null; Status Code: 400; Error Code: null; Request ID: null; Proxy: null)
```

This makes me believe that adding this validation for `webhookTriggersBatchBuild` is the correct thing to do (I won't bother you with adding it for `webhookFilters`, as it's not really the topic of this PR).